// vue3-dropdown-navbar/dist/vue3-dropdown-navbar.mjs
(function(){"use strict";try{if(typeof document<"u"){var t=document.createElement("style");t.appendChild(document.createTextNode("*,:before,:after{--tw-border-spacing-x: 0;--tw-border-spacing-y: 0;--tw-translate-x: 0;--tw-translate-y: 0;--tw-rotate: 0;--tw-skew-x: 0;--tw-skew-y: 0;--tw-scale-x: 1;--tw-scale-y: 1;--tw-pan-x: ;--tw-pan-y: ;--tw-pinch-zoom: ;--tw-scroll-snap-strictness: proximity;--tw-ordinal: ;--tw-slashed-zero: ;--tw-numeric-figure: ;--tw-numeric-spacing: ;--tw-numeric-fraction: ;--tw-ring-inset: ;--tw-ring-offset-width: 0px;--tw-ring-offset-color: #fff;--tw-ring-color: rgb(59 130 246 / .5);--tw-ring-offset-shadow: 0 0 #0000;--tw-ring-shadow: 0 0 #0000;--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;--tw-blur: ;--tw-brightness: ;--tw-contrast: ;--tw-grayscale: ;--tw-hue-rotate: ;--tw-invert: ;--tw-saturate: ;--tw-sepia: ;--tw-drop-shadow: ;--tw-backdrop-blur: ;--tw-backdrop-brightness: ;--tw-backdrop-contrast: ;--tw-backdrop-grayscale: ;--tw-backdrop-hue-rotate: ;--tw-backdrop-invert: ;--tw-backdrop-opacity: ;--tw-backdrop-saturate: ;--tw-backdrop-sepia: }::backdrop{--tw-border-spacing-x: 0;--tw-border-spacing-y: 0;--tw-translate-x: 0;--tw-translate-y: 0;--tw-rotate: 0;--tw-skew-x: 0;--tw-skew-y: 0;--tw-scale-x: 1;--tw-scale-y: 1;--tw-pan-x: ;--tw-pan-y: ;--tw-pinch-zoom: ;--tw-scroll-snap-strictness: proximity;--tw-ordinal: ;--tw-slashed-zero: ;--tw-numeric-figure: ;--tw-numeric-spacing: ;--tw-numeric-fraction: ;--tw-ring-inset: ;--tw-ring-offset-width: 0px;--tw-ring-offset-color: #fff;--tw-ring-color: rgb(59 130 246 / .5);--tw-ring-offset-shadow: 0 0 #0000;--tw-ring-shadow: 0 0 #0000;--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;--tw-blur: ;--tw-brightness: ;--tw-contrast: ;--tw-grayscale: ;--tw-hue-rotate: ;--tw-invert: ;--tw-saturate: ;--tw-sepia: ;--tw-drop-shadow: ;--tw-backdrop-blur: ;--tw-backdrop-brightness: ;--tw-backdrop-contrast: ;--tw-backdrop-grayscale: ;--tw-backdrop-hue-rotate: ;--tw-backdrop-invert: ;--tw-backdrop-opacity: ;--tw-backdrop-saturate: ;--tw-backdrop-sepia: }.dd-nav-container{width:100%}@media (min-width: 640px){.dd-nav-container{max-width:640px}}@media (min-width: 768px){.dd-nav-container{max-width:768px}}@media (min-width: 1024px){.dd-nav-container{max-width:1024px}}@media (min-width: 1280px){.dd-nav-container{max-width:1280px}}@media (min-width: 1536px){.dd-nav-container{max-width:1536px}}.dd-nav-sr-only{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;clip:rect(0,0,0,0);white-space:nowrap;border-width:0}.dd-nav-pointer-events-none{pointer-events:none}.dd-nav-pointer-events-auto{pointer-events:auto}.dd-nav-invisible{visibility:hidden}.dd-nav-absolute{position:absolute}.dd-nav-z-10{z-index:10}.dd-nav-mx-auto{margin-left:auto;margin-right:auto}.dd-nav-my-1{margin-top:.25rem;margin-bottom:.25rem}.dd-nav-my-4{margin-top:1rem;margin-bottom:1rem}.dd-nav-mb-4{margin-bottom:1rem}.dd-nav-ml-1{margin-left:.25rem}.dd-nav-ml-3{margin-left:.75rem}.dd-nav-mr-1{margin-right:.25rem}.dd-nav-mr-2{margin-right:.5rem}.dd-nav-mr-3{margin-right:.75rem}.dd-nav-mt-2{margin-top:.5rem}.dd-nav-block{display:block}.dd-nav-flex{display:flex}.dd-nav-inline-flex{display:inline-flex}.dd-nav-hidden{display:none}.dd-nav-h-0{height:0px}.dd-nav-h-5{height:1.25rem}.dd-nav-h-6{height:1.5rem}.dd-nav-h-9{height:2.25rem}.dd-nav-w-0{width:0px}.dd-nav-w-5{width:1.25rem}.dd-nav-w-6{width:1.5rem}.dd-nav-w-full{width:100%}.dd-nav-flex-col{flex-direction:column}.dd-nav-flex-wrap{flex-wrap:wrap}.dd-nav-items-center{align-items:center}.dd-nav-justify-center{justify-content:center}.dd-nav-justify-between{justify-content:space-between}.dd-nav-divide-y>:not([hidden])~:not([hidden]){--tw-divide-y-reverse: 0;border-top-width:calc(1px * calc(1 - var(--tw-divide-y-reverse)));border-bottom-width:calc(1px * var(--tw-divide-y-reverse))}.dd-nav-divide-gray-100>:not([hidden])~:not([hidden]){--tw-divide-opacity: 1;border-color:rgb(243 244 246 / var(--tw-divide-opacity))}.dd-nav-self-center{align-self:center}.dd-nav-whitespace-nowrap{white-space:nowrap}.dd-nav-rounded{border-radius:.25rem}.dd-nav-rounded-lg{border-radius:.5rem}.dd-nav-border{border-width:1px}.dd-nav-border-0{border-width:0px}.dd-nav-border-x-\\[15px\\]{border-left-width:15px;border-right-width:15px}.dd-nav-border-y{border-top-width:1px;border-bottom-width:1px}.dd-nav-border-b-\\[40px\\]{border-bottom-width:40px}.dd-nav-border-\\[transparent_transparent_\\#00d9ff_transparent\\]{border-color:transparent transparent #00d9ff transparent}.dd-nav-border-gray-100{--tw-border-opacity: 1;border-color:rgb(243 244 246 / var(--tw-border-opacity))}.dd-nav-border-gray-200{--tw-border-opacity: 1;border-color:rgb(229 231 235 / var(--tw-border-opacity))}.dd-nav-border-y-gray-100{--tw-border-opacity: 1;border-top-color:rgb(243 244 246 / var(--tw-border-opacity));border-bottom-color:rgb(243 244 246 / var(--tw-border-opacity))}.dd-nav-bg-blue-500{--tw-bg-opacity: 1;background-color:rgb(59 130 246 / var(--tw-bg-opacity))}.dd-nav-bg-gray-50{--tw-bg-opacity: 1;background-color:rgb(249 250 251 / var(--tw-bg-opacity))}.dd-nav-bg-white{--tw-bg-opacity: 1;background-color:rgb(255 255 255 / var(--tw-bg-opacity))}.dd-nav-p-2{padding:.5rem}.dd-nav-p-4{padding:1rem}.dd-nav-px-2{padding-left:.5rem;padding-right:.5rem}.dd-nav-px-4{padding-left:1rem;padding-right:1rem}.dd-nav-py-1{padding-top:.25rem;padding-bottom:.25rem}.dd-nav-py-2{padding-top:.5rem;padding-bottom:.5rem}.dd-nav-text-sm{font-size:.875rem;line-height:1.25rem}.dd-nav-text-xl{font-size:1.25rem;line-height:1.75rem}.dd-nav-font-medium{font-weight:500}.dd-nav-font-normal{font-weight:400}.dd-nav-font-semibold{font-weight:600}.dd-nav-text-gray-500{--tw-text-opacity: 1;color:rgb(107 114 128 / var(--tw-text-opacity))}.dd-nav-text-gray-700{--tw-text-opacity: 1;color:rgb(55 65 81 / var(--tw-text-opacity))}.dd-nav-text-white{--tw-text-opacity: 1;color:rgb(255 255 255 / var(--tw-text-opacity))}.dd-nav-opacity-0{opacity:0}.dd-nav-shadow{--tw-shadow: 0 1px 3px 0 rgb(0 0 0 / .1), 0 1px 2px -1px rgb(0 0 0 / .1);--tw-shadow-colored: 0 1px 3px 0 var(--tw-shadow-color), 0 1px 2px -1px var(--tw-shadow-color);box-shadow:var(--tw-ring-offset-shadow, 0 0 #0000),var(--tw-ring-shadow, 0 0 #0000),var(--tw-shadow)}.dd-nav-duration-200{transition-duration:.2s}.hover\\:dd-nav-bg-gray-100:hover{--tw-bg-opacity: 1;background-color:rgb(243 244 246 / var(--tw-bg-opacity))}.focus\\:dd-nav-outline-none:focus{outline:2px solid transparent;outline-offset:2px}.focus\\:dd-nav-ring-2:focus{--tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);--tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);box-shadow:var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow, 0 0 #0000)}.focus\\:dd-nav-ring-gray-200:focus{--tw-ring-opacity: 1;--tw-ring-color: rgb(229 231 235 / var(--tw-ring-opacity)) }.dd-nav-dark .dark\\:dd-nav-divide-gray-600>:not([hidden])~:not([hidden]){--tw-divide-opacity: 1;border-color:rgb(75 85 99 / var(--tw-divide-opacity))}.dd-nav-dark .dark\\:dd-nav-border-gray-700{--tw-border-opacity: 1;border-color:rgb(55 65 81 / var(--tw-border-opacity))}.dd-nav-dark .dark\\:dd-nav-bg-gray-700{--tw-bg-opacity: 1;background-color:rgb(55 65 81 / var(--tw-bg-opacity))}.dd-nav-dark .dark\\:dd-nav-bg-gray-800{--tw-bg-opacity: 1;background-color:rgb(31 41 55 / var(--tw-bg-opacity))}.dd-nav-dark .dark\\:dd-nav-bg-gray-900{--tw-bg-opacity: 1;background-color:rgb(17 24 39 / var(--tw-bg-opacity))}.dd-nav-dark .dark\\:dd-nav-text-gray-400{--tw-text-opacity: 1;color:rgb(156 163 175 / var(--tw-text-opacity))}.dd-nav-dark .dark\\:dd-nav-text-white{--tw-text-opacity: 1;color:rgb(255 255 255 / var(--tw-text-opacity))}.dd-nav-dark .dark\\:hover\\:dd-nav-bg-gray-600:hover{--tw-bg-opacity: 1;background-color:rgb(75 85 99 / var(--tw-bg-opacity))}.dd-nav-dark .dark\\:hover\\:dd-nav-bg-gray-700:hover{--tw-bg-opacity: 1;background-color:rgb(55 65 81 / var(--tw-bg-opacity))}.dd-nav-dark .dark\\:hover\\:dd-nav-text-white:hover{--tw-text-opacity: 1;color:rgb(255 255 255 / var(--tw-text-opacity))}.dd-nav-dark .dark\\:focus\\:dd-nav-text-white:focus{--tw-text-opacity: 1;color:rgb(255 255 255 / var(--tw-text-opacity))}.dd-nav-dark .dark\\:focus\\:dd-nav-ring-gray-600:focus{--tw-ring-opacity: 1;--tw-ring-color: rgb(75 85 99 / var(--tw-ring-opacity)) }@media (min-width: 640px){.sm\\:dd-nav-h-10{height:2.5rem}}@media (min-width: 768px){.md\\:dd-nav-absolute{position:absolute}.md\\:dd-nav-my-0{margin-top:0;margin-bottom:0}.md\\:dd-nav-block{display:block}.md\\:dd-nav-hidden{display:none}.md\\:dd-nav-w-44{width:11rem}.md\\:dd-nav-w-auto{width:auto}.md\\:dd-nav-flex-row{flex-direction:row}.md\\:dd-nav-space-x-8>:not([hidden])~:not([hidden]){--tw-space-x-reverse: 0;margin-right:calc(2rem * var(--tw-space-x-reverse));margin-left:calc(2rem * calc(1 - var(--tw-space-x-reverse)))}.md\\:dd-nav-border-0{border-width:0px}.md\\:dd-nav-bg-white{--tw-bg-opacity: 1;background-color:rgb(255 255 255 / var(--tw-bg-opacity))}.md\\:dd-nav-px-4{padding-left:1rem;padding-right:1rem}.md\\:dd-nav-text-sm{font-size:.875rem;line-height:1.25rem}.md\\:dd-nav-font-medium{font-weight:500}.md\\:hover\\:dd-nav-bg-gray-100:hover{--tw-bg-opacity: 1;background-color:rgb(243 244 246 / var(--tw-bg-opacity))}.dd-nav-dark .md\\:dark\\:dd-nav-bg-gray-900{--tw-bg-opacity: 1;background-color:rgb(17 24 39 / var(--tw-bg-opacity))}.dd-nav-dark .md\\:dark\\:hover\\:dd-nav-bg-gray-600:hover{--tw-bg-opacity: 1;background-color:rgb(75 85 99 / var(--tw-bg-opacity))}}")),document.head.appendChild(t)}}catch(d){console.error("vite-plugin-css-injected-by-js",d)}})();
import { defineComponent as _, getCurrentInstance as L, inject as y, openBlock as i, createBlock as $, resolveDynamicComponent as T, unref as p, normalizeProps as N, withCtx as x, createElementBlock as f, createElementVNode as o, renderSlot as m, ref as v, onMounted as B, onUnmounted as R, provide as k, createVNode as U, createTextVNode as A, computed as H, toDisplayString as O, normalizeClass as P, normalizeStyle as V } from "vue";
import { RouterLink as z } from "vue-router";
const j = ["src", "alt"], I = {
  key: 1,
  class: "dd-nav-w-0 dd-nav-h-0 dd-nav-mr-2 dd-nav-border-0 dd-nav-border-x-[15px] dd-nav-border-b-[40px] dd-nav-border-[transparent_transparent_#00d9ff_transparent]"
}, S = { class: "dd-nav-self-center dd-nav-text-xl dd-nav-font-semibold dd-nav-whitespace-nowrap dark:dd-nav-text-white" }, K = /* @__PURE__ */ _({
  __name: "TheDropDownNavbarLogo",
  props: {
    link: { default: "" },
    alt: { default: "Logo" },
    imageUrl: { default: "" },
    native: { type: Boolean, default: !1 }
  },
  setup(l) {
    const a = l, n = L(), t = y("closeAllDropdownMenu"), d = (n == null ? void 0 : n.appContext.components.RouterLink) !== void 0 && !a.native, c = d ? "to" : "href", s = (e) => {
      a.link === "" && e.preventDefault(), t();
    };
    return (e, u) => (i(), $(T(p(d) ? p(z) : "a"), N({
      class: "dd-nav-flex dd-nav-items-center dd-nav-py-2",
      onClick: s,
      [p(c) || ""]: a.link
    }), {
      default: x(() => [
        a.imageUrl ? (i(), f("img", {
          key: 0,
          src: a.imageUrl,
          class: "dd-nav-h-6 dd-nav-mr-3 sm:dd-nav-h-10",
          alt: a.alt
        }, null, 8, j)) : (i(), f("div", I)),
        o("span", S, [
          m(e.$slots, "default")
        ])
      ]),
      _: 3
    }, 16));
  }
}), q = { class: "dd-nav-px-2 dd-nav-bg-white dd-nav-border-gray-200 dark:dd-nav-bg-gray-900 dark:dd-nav-border-gray-700" }, F = { class: "dd-nav-container dd-nav-flex dd-nav-flex-wrap dd-nav-items-center dd-nav-justify-between dd-nav-mx-auto" }, G = /* @__PURE__ */ o("span", { class: "dd-nav-sr-only" }, "open main menu", -1), J = /* @__PURE__ */ o("svg", {
  class: "dd-nav-w-6 dd-nav-h-6",
  "aria-hidden": "true",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ o("path", {
    "fill-rule": "evenodd",
    d: "M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z",
    "clip-rule": "evenodd"
  })
], -1), Q = [
  G,
  J
], W = { class: "dd-nav-flex dd-nav-flex-col dd-nav-flex-wrap dd-nav-p-4 dd-nav-mt-2 dd-nav-mb-4 dd-nav-border dd-nav-border-gray-100 dd-nav-rounded-lg dd-nav-bg-gray-50 md:dd-nav-flex-row md:dd-nav-space-x-8 md:dd-nav-my-0 md:dd-nav-text-sm md:dd-nav-font-medium md:dd-nav-border-0 md:dd-nav-bg-white dark:dd-nav-bg-gray-800 md:dark:dd-nav-bg-gray-900 dark:dd-nav-border-gray-700" }, rd = /* @__PURE__ */ _({
  __name: "TheDropDownNavbar",
  setup(l, { expose: a }) {
    const n = v(), t = v(), d = v([]), c = () => {
      n.value && n.value.classList.toggle("dd-nav-hidden");
    }, s = (r) => {
      d.value.push(r);
    }, e = () => {
      d.value.forEach((r) => r());
    }, u = (r) => {
      var g, h;
      r.target.getAttribute("no-close-others") === null && !((g = n.value) != null && g.contains(r.target)) && !((h = t.value) != null && h.contains(r.target)) && e();
    };
    return B(() => {
      n.value && (n.value.classList.add("dd-nav-hidden"), n.value.classList.remove("dd-nav-invisible")), window.addEventListener("click", u);
    }), R(() => {
      window.removeEventListener("click", u);
    }), k("addCloseDropdownMenuCallback", s), k("closeAllDropdownMenu", e), a({
      closeAllDropdownMenu: e
    }), (r, g) => (i(), f("nav", q, [
      o("div", F, [
        o("div", {
          ref_key: "logo",
          ref: t
        }, [
          m(r.$slots, "logo", {}, () => [
            U(K, null, {
              default: x(() => [
                A("Vue3 DropDown Navbar")
              ]),
              _: 1
            })
          ])
        ], 512),
        o("button", {
          type: "button",
          class: "dd-nav-inline-flex dd-nav-items-center dd-nav-p-2 dd-nav-h-9 dd-nav-my-4 dd-nav-ml-3 dd-nav-mr-1 dd-nav-text-sm dd-nav-text-gray-500 dd-nav-rounded-lg md:dd-nav-hidden hover:dd-nav-bg-gray-100 focus:dd-nav-outline-none focus:dd-nav-ring-2 focus:dd-nav-ring-gray-200 dark:dd-nav-text-gray-400 dark:hover:dd-nav-bg-gray-700 dark:focus:dd-nav-ring-gray-600",
          onClick: c
        }, Q),
        o("div", {
          id: "dropdown-root",
          class: "dd-nav-invisible dd-nav-w-full md:dd-nav-block md:dd-nav-w-auto",
          ref_key: "navbarDropdown",
          ref: n
        }, [
          o("ul", W, [
            m(r.$slots, "default")
          ])
        ], 512)
      ])
    ]));
  }
}), X = /* @__PURE__ */ o("svg", {
  class: "dd-nav-w-5 dd-nav-h-5 dd-nav-ml-1",
  "aria-hidden": "true",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ o("path", {
    "fill-rule": "evenodd",
    d: "M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z",
    "clip-rule": "evenodd"
  })
], -1), Y = {
  class: "dd-nav-py-2 dd-nav-text-sm dd-nav-text-gray-700 dark:dd-nav-text-gray-400",
  "aria-labelledby": "dropdownLargeButton"
}, ld = /* @__PURE__ */ _({
  __name: "TheDropDownMenu",
  props: {
    text: null,
    closeOthers: { type: Boolean, default: !0 }
  },
  setup(l, { expose: a }) {
    const n = l, t = v(!1), d = v(), c = v(), s = v(0), e = v(!1), u = H(() => {
      var b, M, C;
      const w = (C = (M = (b = d.value) == null ? void 0 : b.parentNode) == null ? void 0 : M.parentNode) == null ? void 0 : C.parentNode;
      return (w == null ? void 0 : w.id) === "dropdown-root";
    }), r = y("closeAllDropdownMenu"), g = () => {
      d.value && (n.closeOthers && u.value && !e.value && r(), d.value.classList.toggle("dd-nav-opacity-0"), d.value.classList.toggle("dd-nav-pointer-events-auto"), e.value = !e.value);
    }, h = () => {
      e.value && g();
    }, E = y(
      "openParentDropdownMenu",
      () => {
      }
    ), D = () => {
      t.value = !0, E(), d.value && !e.value && (d.value.classList.toggle("dd-nav-opacity-0"), d.value.classList.toggle("dd-nav-pointer-events-auto"), e.value = !0), setTimeout(() => {
        t.value = !1;
      });
    };
    return y("addCloseDropdownMenuCallback")(() => {
      t.value || h();
    }), B(() => {
      d.value && (s.value = d.value.clientHeight, d.value.style.height = "0px");
    }), k("openParentDropdownMenu", D), a({
      openDropdownMenu: D,
      closeDropdownMenu: h
    }), (w, b) => (i(), f("li", null, [
      o("button", {
        class: "dd-nav-flex dd-nav-items-center dd-nav-justify-between dd-nav-w-full dd-nav-py-2 dd-nav-px-4 dd-nav-font-medium dd-nav-text-gray-700 dd-nav-rounded hover:dd-nav-bg-gray-100 md:dd-nav-px-4 md:dd-nav-border-0 md:hover:dd-nav-bg-gray-100 dark:dd-nav-text-gray-400 dark:hover:dd-nav-text-white dark:focus:dd-nav-text-white dark:dd-nav-border-gray-700 dark:hover:dd-nav-bg-gray-700 md:dark:hover:dd-nav-bg-gray-600",
        onClick: g,
        ref_key: "dropdownTrigger",
        ref: c
      }, [
        A(O(n.text) + " ", 1),
        X
      ], 512),
      o("div", {
        ref_key: "dropdownMenu",
        ref: d,
        class: P(["dd-nav-opacity-0 dd-nav-pointer-events-none dd-nav-duration-200 dd-nav-z-10 dd-nav-w-full dd-nav-font-normal dd-nav-bg-white dd-nav-divide-y dd-nav-divide-gray-100 dd-nav-rounded-lg dd-nav-shadow md:dd-nav-w-44 dark:dd-nav-bg-gray-700 dark:dd-nav-divide-gray-600", { "md:dd-nav-absolute": p(u) }]),
        style: V({
          height: e.value ? "unset" : "0px",
          minHeight: e.value ? `${s.value}px` : "unset"
        })
      }, [
        o("ul", Y, [
          m(w.$slots, "default")
        ])
      ], 6)
    ]));
  }
}), sd = /* @__PURE__ */ _({
  __name: "TheDropDownItem",
  props: {
    link: { default: "" },
    native: { type: Boolean, default: !1 }
  },
  setup(l) {
    const a = l, n = L(), t = y("closeAllDropdownMenu"), d = (n == null ? void 0 : n.appContext.components.RouterLink) !== void 0 && !a.native, c = d ? "to" : "href", s = (e) => {
      a.link === "" && e.preventDefault(), t();
    };
    return (e, u) => (i(), f("li", null, [
      (i(), $(T(p(d) ? p(z) : "a"), N({
        class: "dd-nav-block dd-nav-px-4 dd-nav-py-2 hover:dd-nav-bg-gray-100 dark:dd-nav-text-gray-400 dark:hover:dd-nav-bg-gray-600 dark:hover:dd-nav-text-white",
        onClick: s,
        [p(c) || ""]: a.link
      }), {
        default: x(() => [
          m(e.$slots, "default")
        ]),
        _: 3
      }, 16))
    ]));
  }
}), Z = (l, a) => {
  const n = l.__vccOpts || l;
  for (const [t, d] of a)
    n[t] = d;
  return n;
}, dd = {}, nd = { class: "dd-nav-py-1 dd-nav-my-1 dd-nav-border-y dd-nav-border-y-gray-100" };
function ed(l, a) {
  return i(), f("li", nd, [
    o("ul", null, [
      m(l.$slots, "default")
    ])
  ]);
}
const vd = /* @__PURE__ */ Z(dd, [["render", ed]]);
export {
  vd as TheDropDownDivideBlock,
  sd as TheDropDownItem,
  ld as TheDropDownMenu,
  rd as TheDropDownNavbar,
  K as TheDropDownNavbarLogo
};
